daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs

```diff
@@ -711,6 +753,26 @@ getProjectName = do
     requiredE "Project must have a name" $
         queryProjectConfigRequired ["name"] projectConfig
 
+getProjectParties :: IO [String]
+getProjectParties = do
+    projectConfig <- getProjectConfig
+    requiredE "Project must have parties listed" $
+        queryProjectConfigRequired ["parties"] projectConfig
+
+-- TODO: `daml sandbox` should also consult the config for the ledger-port
+-- Have daml-helper wrap the `sandbox` command
+getProjectLedgerPort :: IO Int
+getProjectLedgerPort = do
+    projectConfig <- getProjectConfig
+    -- TODO: remove default; insist ledger-port is in the config ?!
+    defaulting 6865 $ queryProjectConfig ["ledger-port"] projectConfig
```

Comment:
I think it would be better to use `queryProjectConfig ["deploy", "port"]` here, and `queryProjectConfig ["deploy", "host"]` below.

Reply:
Ok. Do we already make use of this kind hierarchical config anywhere else?

Reply:
Not yet, but it's probably a bad thing that the project config file is so unstructured at the moment. E.g. build config really ought to be separate from project metadata.

Hierarchy is used in the SDK config to determine commands that are available, but it doesn't use `querySdkConfig` to get the individual command data (it couldn't, since it's a list).

Reply:
Here's the right way to do it:

Define a `DeployConfig` data structure. Implement `FromJSON` for it. Then use `queryProjectConfig ["deploy"]` to get it.

Reply:
Again,  we can postpone this kind of restructuring to later.